src/linker/Linker/Annotations.cs

```diff
@@ -56,6 +56,7 @@ public partial class AnnotationStore
 		protected readonly Dictionary<MethodDefinition, List<MethodDefinition>> base_methods = new Dictionary<MethodDefinition, List<MethodDefinition>> ();
 		protected readonly Dictionary<AssemblyDefinition, ISymbolReader> symbol_readers = new Dictionary<AssemblyDefinition, ISymbolReader> ();
 		readonly Dictionary<MethodDefinition, LinkerAttributesInformation> method_linker_attributes = new Dictionary<MethodDefinition, LinkerAttributesInformation> ();
+		readonly Dictionary<FieldDefinition, LinkerAttributesInformation> field_linker_attributes = new Dictionary<FieldDefinition, LinkerAttributesInformation> ();
```

Comment:
I already though about this - I think a better solution is to key the Dictionary with ICustomAttributeProvider and make the cache general. I don't see much value in having special code in the cache which would know which attributes apply to which types (it doesn't need to for anything really).